modular_skyrat/modules/modular_items/lewd_items/code/lewd_arousal/arousal.dm

```diff
@@ -49,6 +49,17 @@
 
 	return TRUE
 
+/mob/living/carbon/human/Initialize(mapload)
+	. = ..()
+	RegisterSignal(src, COMSIG_LIVING_LIFE, PROC_REF(on_life_tick))
```

Comment:
if keep_arousal_on_goal isn't used for anything else you could simply use RegisterSignal and UnregisterSignal in toggling arousal as the way you control the running of this